swift_scality_backend/server.py

```diff
@@ -60,63 +70,141 @@ def setup(self, conf):
+            if sproxyd_conn_timeout is not None else None
+        self._read_timeout = float(sproxyd_read_timeout) \
+            if sproxyd_read_timeout is not None else None
+
+        sp_path = \
+            swift_scality_backend.policy_configuration.DEFAULT_CONFIGURATION_PATH
+        self.logger.info('Reading storage policy configuration from %r', sp_path)
+        try:
+            with open(sp_path, 'r') as fd:
+                self.logger.info('Parsing storage policy configuration')
+                self._policy_configuration = \
+                    swift_scality_backend.policy_configuration.Configuration.from_stream(fd)
+        except IOError as exc:
+            if exc.errno == errno.ENOENT:
+                self.logger.info(
+                    'No storage policy configuration found at %r', sp_path)
+                self._policy_configuration = None
+            else:
+                self.logger.exception(
+                    'Failure while reading storage policy configuration '
+                    'from %r', sp_path)
+                raise
+
+        location_preferences = conf.get('scality_location_preferences')
+        if location_preferences is None:
+            self._location_preferences = None
+        else:
+            self._location_preferences = \
+                swift_scality_backend.utils.split_list(location_preferences)
+
+    def _get_client_for_policy(self, policy_idx):
+        '''Retrieve or create an Sproxyd client for a given storage policy
+
+        :param policy_idx: Policy identifier
+        :type policy_idx: `int`
+        :return: Sproxyd client which can be used for requests in the given
+                 policy
+        :rtype: `scality_sproxyd_client.sproxyd_client.SproxydClient`
+
+        :raise RuntimeError: No policies configured
+        '''
+
+        sproxyd_client_kwargs = {
+            'logger': self.logger,
+        }
+        if self._conn_timeout is not None:
+            sproxyd_client_kwargs['conn_timeout'] = self._conn_timeout
+        if self._read_timeout is not None:
+            sproxyd_client_kwargs['read_timeout'] = self._read_timeout
+
+        if policy_idx not in self._clients:
+            collection = None
+
+            if policy_idx == 0:
+                endpoints = self._policy_0_urls
+
+                client = scality_sproxyd_client.sproxyd_client.SproxydClient(
+                    endpoints, **sproxyd_client_kwargs)
+
+                collection = swift_scality_backend.diskfile.ClientCollection(
+                    read_clients=[client], write_clients=[client])
+            else:
+                if not self._policy_configuration:
+                    raise RuntimeError(
+                        'No storage policy configuration found, but request '
+                        'for policy %r' % policy_idx)
+
+                policy = self._policy_configuration.get_policy(policy_idx)
+
+                clients = {}
```

Comment:
What's the purpose of this variable ? It looks like an accumulator, or some kind of memory. I'd like a small code comment.